contracts/governance/Vesting/VestingRegistryLogic.sol

```diff
@@ -289,6 +311,46 @@ contract VestingRegistryLogic is VestingRegistryStorage {
         return vestings[uid].vestingAddress;
     }
 
+    /**
+     * @dev check if the specific vesting address is team vesting or not
+     * @dev read the vestingType from vestingCreationAndTypes storage
+     *
+     * @param _vestingAddress address of vesting contract
+     *
+     * @return true for teamVesting, false for normal vesting
+     */
+    function isTeamVesting(address _vestingAddress) external view returns (bool) {
+        return (vestingCreationAndTypes[_vestingAddress].isSet &&
+            vestingCreationAndTypes[_vestingAddress].vestingType ==
+            uint256(VestingType.TeamVesting));
+    }
+
+    /**
+     * @dev setter function to register existing vesting contract to vestingCreationAndTypes storage
+     * @dev need to set the function visilibty to public to support VestingCreationAndTypeDetails struct as parameter
+     *
+     * @param _vestingAddresses array of vesting address
+     * @param _vestingCreationAndTypes array for VestingCreationAndTypeDetails struct
+     */
+    function registerVestingToVestingCreationAndTypes(
```

Comment:
current design requires updating all the vesting contracts which will be costly  
also a script getting all the vestings, splitting by chunks (gas limit) and updating vestings is needed for this   
but the structure `vestingCreationAndTypeDetails` and the new mapping to store vestings details was introduced for vestings cancelling exclusively(?),   
i suggest an alternative design:  
remove `vestingCreationAndTypeDetails` and use a VestingRegistry's   
```
 function getTeamVesting(
        address _tokenOwner,
        uint256 _cliff,
        uint256 _duration,
        uint256 _vestingCreationType
    )
``` 
    option 1: (more params but less constly): change Staking `cancelTeamVesting(...)` params to the above params and call `getTeamVesting()`  
    option 2: (less params but more constly): change `isTeamVesting()` to get vesting contract, pull the above params from the vesting, call `getTeamVesting` with the params, check address is the same as the vesting address param  
@ororopickpocket - wdyt?

Reply:
Registering multiple vesting contracts can be done within 1 transaction since the register function accepting the array of addresses.

Here is the roughly estimation about the number:
Base gas usage for registering 1 vesting contract: 57501
Every additional contracts will add up the gas usage by 27249

Do we have the list contains our vesting addresses @ororopickpocket  ?

But for your reference the price for registering 10 vesting addresses will be:
Gas used: 302764
Gas price: 0.06 GWEI
The number will be ~ 0.00001816584 RBTC

Reply:
we don't have a list, we have to read them from the registry. 
The cost for registering them is okay. I am in favor of setting the details for all existing contracts.

Reply:
ok. seems not to be that much to process all vestings.